modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManager.java

```diff
@@ -278,8 +279,16 @@ public SCIMResponse listWithGET(UserManager userManager, String filter, int star
                 List<Object> tempList = userManager.listGroupsWithGET(rootNode, startIndex, count,
                         sortBy, sortOrder, requiredAttributes);
 
-                totalResults = (int) tempList.get(0);
-                tempList.remove(0);
+                if (tempList == null) {
+                    tempList = Collections.emptyList();
+                }
+
+                try {
+                    totalResults = (int) tempList.get(0);
+                    tempList.remove(0);
+                } catch (IndexOutOfBoundsException | ClassCastException ex) {
```

Comment:
shall we add a error/debug log here?

Reply:
Fixed the review comments with https://github.com/wso2/charon/pull/129

Reply:
a warning would be recommendable since the developer should be informed that the totalResults field is defaulted to the size of the returned list